src/App.tsx

```diff
@@ -1,25 +1,84 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
+type TypeOfState = {
+  PageChange: string;
+  CurrentPage: number;
+};
+
+enum SelectFields {
+  colomns3 = '3',
+  colomns5 = '5',
+  colomns10 = '10',
+  colomns20 = '20',
+}
+
+function PrepearedItems(
+  oldItems: string[],
+  { PageChange, CurrentPage }: TypeOfState,
+) {
+  let prepearedItems: string[] = [...oldItems];
+
+  if (CurrentPage) {
+    const extraElements = (CurrentPage - 1) * Number(PageChange);
+
+    prepearedItems = prepearedItems.slice(extraElements);
+  }
+
+  switch (PageChange) {
+    case SelectFields.colomns3:
+      return prepearedItems.slice(0, 3);
+    case SelectFields.colomns5:
+      return prepearedItems.slice(0, 5);
+    case SelectFields.colomns10:
+      return prepearedItems.slice(0, 10);
+    case SelectFields.colomns20:
+      return prepearedItems.slice(0, 20);
+
+    default:
+      return prepearedItems;
+  }
+}
+
 export const App: React.FC = () => {
+  const [PageChange, onPageChange] = useState('5');
+  const [Pages, setPages] = useState(9);
+  const [CurrentPage, setCurrentPage] = useState(1);
```

Comment:
You need only these two states
```suggestion
  const [perPage, setPerPage] = useState(5);
  const [currentPage, setCurrentPage] = useState(1);
```